test/test_mxnet.py

```diff
@@ -0,0 +1,324 @@
+# Copyright 2018 Uber Technologies, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import itertools
+import subprocess
+import mxnet as mx
+import unittest
+import numpy as np
+import horovod.mxnet as hvd
+
+
+# Currently, when we build and install Horovod with MXNet pip package, we need
+# to do it on the same OS which MXNet pip package is built. Otherwise, we will
+# hit segmentation fault when running MXNet unit tests, which we suspect is due
+# to different versions of libc library. While we are investigating the root
+# cause and remove this requirement, we will skip unit tests for MXNet
```

Comment:
We should figure it out before landing, as the majority of users will not be on later versions of OS.

Reply:
We have been investigating this issue for a while but have not figured out the root cause yet.

The MXNet pip package is built on Ubuntu 14.04 with gcc version 4.8.4. If we install MXNet using pip package and then pip install horovod on a OS with gcc version higher than 5 (Ubuntu 16.04: 5.4.0, Ubuntu 18.04: 7.3.0, Debian/stretch: 6.3.0), we hit the segfault while running MXNet unit tests and examples. We suspect that it may be related to different gcc versions (and libstdc++) we used to build MXNet and Horovod on different OSes.

If we build MXNet from source and then install horovod on a newer OS (Ubuntu 16.04, Ubuntu 18.04, Debian/Stretch, Debian/Buster), we do not see any issue running MXNet unit tests and examples (maybe because we compile MXNet and horovod using the same gcc).

We do not want to block this PR while we are investigating this issue. That's why we skip unittests and example on OSes other than jessie in the CI. Another option is to build MXNet from source in the CI but it will take longer which might not be a very good idea. Please suggest.